fcrepo4/fcrepo4
# Make all LDP-Rs versionable by default

**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-2890

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)
 https://jira.duraspace.org/browse/FCREPO-2889

# What does this Pull Request do?
This PR makes all LDP-Rs versionable by default.   Now the  `Link: <http://mementoweb.org/ns#OriginalResource>; rel="type"`  header is simply ignored on creation of PUTs and POSTs.  The related integration tests were removed or modified.

# How should this be tested?
Create  resources without the versioning link header and verify that all resources are versionable regardless of whether the versioning header is present.
```
# rdf source
curl -i -u fedoraAdmin:fedoraAdmin -X POST http://localhost:8080/rest -H "Slug: test1-rdf"; 
curl -i -u fedoraAdmin:fedoraAdmin -X GET http://localhost:8080/rest/test1-rdf
# non rdf 
curl -i -u fedoraAdmin:fedoraAdmin -X POST http://localhost:8080/rest -H "Slug: test1-nr" -H "Content-Type: text/plain"; 
curl -i -u fedoraAdmin:fedoraAdmin -X GET http://localhost:8080/rest/test1-nr
# rdf source with header
curl -i -u fedoraAdmin:fedoraAdmin -X POST http://localhost:8080/rest -H "Slug: test2-rdf" -H "Link: <http://mementoweb.org/ns#OriginalResource>; rel=\"type\"";  
curl -i -u fedoraAdmin:fedoraAdmin -X GET http://localhost:8080/rest/test2-rdf
# non rdf with header
curl -i -u fedoraAdmin:fedoraAdmin -X POST http://localhost:8080/rest -H "Slug: test2-nr" -H "Content-Type: text/plain"  -H "Link: <http://mementoweb.org/ns#OriginalResource>; rel=\"type\""; 
curl -i -u fedoraAdmin:fedoraAdmin -X GET http://localhost:8080/rest/test2-nr
```

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo4/committers
